### ISSUE TITLE
Conformer ASR model gives prediction over a few words

### ISSUE THREAD

**Issue by thechuong98** [speechbrain/speechbrain#979]
Hi, I am training the `Conformer` on a `Vietnamese dataset` (around 100 hours of speech). I have trained a `Sentencepiece tokenizer` (5000_unigram) with the dataset, and it looks like this 

![image](https://user-images.githubusercontent.com/60946216/132705800-69d2e8d0-5ee7-4155-86bf-9ac25af3420e.png)

Then I trained the ASR model with the recipe `speechbrain/recipes/LibriSpeech/ASR/transformer/train.py` and then re-config the conformer_small.yaml (I only change the datasets csv and using a Vietnamese 5000 unigram tokenizer, other settings left unchanged).
Here is what I got after `45 epochs` of training and The `WER stays ~100` after 45 epochs.

![image](https://user-images.githubusercontent.com/60946216/132706919-5b7b26e5-09ca-4b58-a96f-79b7da46c6e6.png)

I have also tried training on `LibriSpeech train-clean-100` (with the conformer recipe) and WER decreases real fast

![image](https://user-images.githubusercontent.com/60946216/132707760-13cf90f4-2e5d-4ceb-9a13-fa019d0f46e0.png)

The predictions on the test set (Vietnamese model) `spread over only a few words` :

![image](https://user-images.githubusercontent.com/60946216/132708190-d8015315-185e-437f-acb9-e60652a542f6.png)

Has anyone faced the same issue when training in other languages datasets? Should I keep training or what should I do? Thanks a lot

**Comment by thechuong98** [comment]
Here is the output of S2STransformerBeamSearch that I get by running `self.hparams.valid_search(enc_out.detach(), wav_lens)`  
while the model is training
```
(Pdb) self.hparams.valid_search(enc_out.detach(), wav_lens)
([[50, 16], [50, 16], [50, 16], [50, 16], [50, 16], [50, 16], [50, 16], [50, 16], [50, 16], [50, 16], [50, 16], [50, 16], [50, 16], [50, 16], [50, 16], [50, 16]], tensor([[-13.9380],
        [-13.9920],
        [-13.8977],
        [-13.9881],
        [-13.9300],
        [-13.9235],
        [-13.9532],
        [-13.8694],
        [-13.9957],
        [-13.9751],
        [-13.9338],
        [-13.9440],
        [-14.0289],
        [-14.0106],
        [-13.9450],
        [-13.9525]], device='cuda:0', grad_fn=<TopkBackward>))
```

**Comment by mravanelli** [comment]
Sometimes making a transformer-based model converging is challenging. Did you try with a standard transformer instead? An alternative that I might suggest is doing pretraining with wav2vect (it normally helps even on different languages). You can find an example in the current development branch (see /recipes/LibriSpeech/ASR/ctc). The issue is that the ctc+wav2vect system can be very memory demanding.

**Comment by thechuong98** [comment]
Thank you, I tried a smaller version of standard transformer and it's now converging :D
